### PR TITLE
Fixes #229: Pinning xarray to avoid issues with underlying call to pa…

### LINKED ISSUE #229: Need to pin (temporarily) to an older version of xarray
* IceNet version: 0.2.7
* Python version: 3.8.16

### Description

`icenet_data_era5` is failing with

```
[14-03-24 22:42:18 :ERROR   ] - Thread failure: __init__() got an unexpected keyword argument 'base'
Traceback (most recent call last):
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/icenet/data/interfaces/downloader.py", line 226, in download
    future.result()
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/icenet/data/interfaces/downloader.py", line 302, in _single_download
    self.postprocess(var, latlon_path)
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/icenet/data/interfaces/cds.py", line 235, in postprocess
    da = da.sortby("time").resample(time='1D').mean()
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/xarray/core/dataarray.py", line 6666, in resample
    return self._resample(
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/xarray/core/common.py", line 993, in _resample
    grouper = pd.Grouper(
  File "/data/hpcdata/users/jambyr/miniconda3/envs/icenet-new/lib/python3.8/site-packages/pandas/core/resample.py", line 1663, in __init__
    super().__init__(freq=freq, axis=axis, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'base'
```

https://github.com/pydata/xarray/issues/2385 is the cause, [v2023.01.0](https://github.com/pydata/xarray/releases/tag/v2023.01.0) dropped support for python3.8 which is likely the underlying issue

### What I Did

Running `run_era5_forecast.sh` from the pipeline

### PR DESCRIPTION
**Pull request by JimCircadian**
…necessary to fix the broken operational forecasts. Already dealt with in 0.3 dev

### CODE REVIEW

**Review by bnubald — CHANGES_REQUESTED** [review]
Pinning xarray version still causes error.

Cause of #229 was Pandas > 2.0.0 (see https://github.com/icenet-ai/icenet/issues/229#issuecomment-2003816509)

Also, not strictly necessary, but would prefer if if pinning was set for specifically just for 3.8 to not be too restrictive for future versions.

```
pandas<2; python_version == '3.8'
xarray[io]<2023.2.0; python_version == '3.8'
xarray[io]; python_version > '3.8'
```

File: requirements.txt [hunk 1]
```diff
@@ -22,4 +22,4 @@ tensorboard
 tensorflow
 tensorflow-probability
 wheel
-xarray[io]
+xarray[io]<2013.1.0
```
[the remark this answers is not in the document]
Comment (JimCircadian) [reply requirements.txt]: Thanks for picking that up that version "typo" Bryn, I was speed reading the release notes so must have lost track.  Updated the changes, though I'm not keen on pinning solely for specific python versions as it makes it more likely we develop against newer features. 

I think this makes the case for eventual dropping of 3.8, but let's pick that up for the 0.3 branch if that's OK?

**Comment by JimCircadian** [comment]
> Pinning xarray version still causes error.
> 
> Cause of #229 was Pandas >= 2.0.0 (see [#229 (comment)](https://github.com/icenet-ai/icenet/issues/229#issuecomment-2003816509))
> 
> Also, not strictly necessary, but would prefer if if pinning was set specifically just for 3.8, to not be too restrictive for future versions.
> 
> ```
> pandas<2; python_version == '3.8'
> xarray[io]<2023.2.0; python_version == '3.8'
> xarray[io]; python_version > '3.8'
> ```

Updated the changes, though I'm not keen on pinning solely for specific python versions as it makes it more likely we develop against newer features. 

I think this makes the case for eventual dropping of 3.8, but let's pick that up for the 0.3 branch if that's OK?

File: requirements.txt [hunk 2]
```diff
@@ -22,4 +23,4 @@ tensorboard
 tensorflow
 tensorflow-probability
 wheel
-xarray[io]
+xarray[io]<2013.2.0
```
Comment (bnubald) [inline requirements.txt:26]: Sorry James, still a typo, should be `xarray[io]<2023.2.0`, change being the year.

**Review by bnubald — CHANGES_REQUESTED** [review]
Still a typo.

File: requirements.txt (hunk 2 above)
Comment (JimCircadian) [reply requirements.txt]: Thanks @bnubald, just checking you're on it 😛

**Comment by bnubald** [comment]
> > Pinning xarray version still causes error.
> > Cause of #229 was Pandas >= 2.0.0 (see [#229 (comment)](https://github.com/icenet-ai/icenet/issues/229#issuecomment-2003816509))
> > Also, not strictly necessary, but would prefer if if pinning was set specifically just for 3.8, to not be too restrictive for future versions.
> > ```
> > pandas<2; python_version == '3.8'
> > xarray[io]<2023.2.0; python_version == '3.8'
> > xarray[io]; python_version > '3.8'
> > ```
> 
> Updated the changes, though I'm not keen on pinning solely for specific python versions as it makes it more likely we develop against newer features.
> 
> I think this makes the case for eventual dropping of 3.8, but let's pick that up for the 0.3 branch if that's OK?

👍🏼

**Review by bnubald — APPROVED** [review]

**Pull request merged**